### PR TITLE
Add ouput_device parameter to get_stats

### PR DESCRIPTION
**Pull request by witoong623**
Hello. I'm new to using `smp.metrics` (previously I used metric functions from utils), but I would like to propose a change, which is adding `output_device` parameter to `get_stats` function.

I assume that people use this function in `'multiclass'` mode, then follow by function such as `iou_score`.
The problem may arise if people use a function from `torch.distributed` such as `all_reduce` on `nccl` backend, to average metric values across processes, because `all_reduce` function requires tensors to be on GPU, but in the case of multiclass, `get_stats` create zeros tensors with default device option, which puts it in CPU, and later return them, so the device of returned tensors is CPU.

I know that I can move a result stats tensor to GPU and it will be fine, but isn't it more efficient to specify the return type from the beginning? Especially when the output and target tensors are on the returned device already (I don't really know if this is true or not😅).

So, I create this PR to add parameter named `output_device` to `get_stats` function so that `torch.zeros` can create tensor on specified device, and some tests to ensure that it doesn't break existing code.

In the case of binary and multilabel mode, I just move the result tensors to specified device because the result will be on the same device as input already.

### CODE REVIEW

**Comment by qubvel** [comment]
Hi @witoong623 

Thanks for the PR, this looks interesting, not sure if tests will work on github actions 😅 (there is no GPU I guess)

**Comment by witoong623** [comment]
Hello @qubvel, thank you for the awesome library!

What should I do in this case? I think I can add skip condition on the tests that use GPU.
